DEID-2180 Audio Redaction Script and Sample Files

AmirPAI opened the pull request:
## Purpose
The purpose of this PR is to add Audio Redaction demo notebook to help users with a notebook that shows how to process audio files with our system.
https://privateai.atlassian.net/browse/DEID-2180

## Description
**What has changed?**
* Added a ipynb script for processing Audio files
* Added sample files (mp3, docx, pdf, csv) Original files and redacted files.


**Screenshots?**

## Run Instructions
**Anything needed to run the code?**

 ### Repo specific

## Wishes
**Anything you want reviewers to focus on?**

**Any specific part you want to receive feedback on?**
@mikeyoungPAI
## Checklist
* [ ]  No linting error?
* [ ]  No dead code?

letmerecall commented on python/examples/Redact_Files.ipynb:
Something like a table of contents that tells us what's in the notebook will be nice to have on top.

AmirPAI replied:
Ok

letmerecall commented on python/structured_data/sample_data/Medical-Report-Audio-Unredacted.mp3:
I don't think it belongs to `strucured_data` folder 🤔, strucutred data are thing like table, forms, etc.

I prefere we put the sample data in `data` folder & outputs in `output` folder.

AmirPAI commented:
@letmerecall done

letmerecall reviewed (changes requested):
Can you please make `input_file_path` & `output_file_path` variables in each file's code cell and put them on the top of that cell? Something like this:

```python
input_docx_path = 'data/PAI_SYNTH_EN_doctor-note.doc'
output_docx_path = 'output/PAI_SYNTH_EN_doctor-note_redacted.doc'

with open(input_docx_path, 'rb') as input_file:
    file_data = base64.b64encode(input_file.read()).decode()

response = client.process_files_base64(
    request_object={
        'file': {
            'data': file_data,
            'content_type': 'application/msword'
        }
    }
)

print('***** PROCESSED TEXT *****')
print(response.processed_text)
print('***** REDACTED TEXT *****')
print(json.dumps(response.entities, indent=2))

with open(output_docx_path, 'wb') as redacted_docx:
  redacted_docx.write(base64.b64decode(response.processed_file))

```

Similarly for all other file types.


Otherwise looks good to me.

AmirPAI commented:
@letmerecall made the changes 👍.

letmerecall reviewed (approved):
Looks good to me! I'd wait for @yaysummeriscoming to have a look.

yaysummeriscoming reviewed (changes requested):
Finally got a chance to look at this, sorry for the delay. Overall LGTM, just minor comments. This covers more than just audio files, but I'm fine with that. Comments:

- We should make it easy for the user to find the content type for other filetypes. For each example, we should include a mention "For other filetypes, please set the content type as per the docs https://docs.private-ai.com/files/"
- Update "demo" to "community" in get an API key comment: "Get a free demo API key here: https://portal.private-ai.com/"
- We should remove CSV.. We aren't too happy with how that performs ATM

AmirPAI commented:
@yaysummeriscoming Added the comments and made the changes👍.

yaysummeriscoming commented:
Thanks @AmirPAI, I was taking a final look and I noticed the DOCX file is actually a DOC file? I'd appreciate it if we could have a DOCX file please, DOC doesn't work quite as well and is a pretty old file format now.

Also could you remove unneeded files from the input and output directories please? I see the CSV files are still in, for example.

I took a look at the redacted PDF file, that looks pretty good - could you make sure that the redacted DOCX and audio files also look good please? You can ask @Irina-Presnyakova for some advice on what constitutes good performance here.

AmirPAI commented:
Modified notebook script to process DOCX file instead and removed extra files.
Audio and DOCX are being redacted pretty good too. Though Audio redaction is taking significantly larger time (>1min compared to PDF and DOCX), it might be because of its nature and the ASR processing. All good.

yaysummeriscoming reviewed (approved):
LGTM now, thanks @AmirPAI. Yeah long audio processing time is a known limitation. Audio just takes a lot of compute, we're working on a GPU port to solve this

Irina-Presnyakova commented:
**Audio:** good redaction overall; some peculiarities: the first 11 seconds of audio contain NAME, NAME_MED_PROF, ORG, MED_PROCESS, DATE in a rapid succession and the "beeps" partially cover non-PII speech, as well. 
Later on, adjacent speech is also covered up, e.g., the sentence "Have you ever had migraines" (26-27s) is completely covered by a beep (only the last word is PII).
At 31 s, there is a partial miss, in the phrase "Any panic attacks?" you can still hear the word "panic," so the beep starts a bit too late.  

**docx:** no issues, no misses, the formatting is preserved (in the header, ORG is the same font as surrounding unredacted text)

**pdf**: very decent redaction overall, just a few of misses:
- In the header, "Fathers mental health" (name of the program/ORG) is not redacted, but it appears to be a part of the logo, so logo redaction will take care of that? 
- in the lower part of the pdf, the handwritten phrase "disrupted sleep/insomnia" is left unredacted (we label "insomnia" as CONDITION)  
- second occurrence of the typed word "psychiatric" in the lower part is left unredacted (it is redacted earlier in the same document)
- at the very bottom, "The Perinatal Mental Heath Program" and "FMH" (Fathers mental health) is left unredacted (could be ORG for both? Or if they are treated as PRODUCT then we expect them to be unredacted)
- 
@yaysummeriscoming @AmirPAI

AmirPAI commented:
Thanks for such a detailed report on that @Irina-Presnyakova! 
Sure, we should be making the audio redaction's timing precise (timestamp precision so that only PII redact not the neighbouring words).
Logo detection will come up soon.
Happy to know that overall everything's good. Merging it!
